test/functional/foreman_statistics/statistics_controller_test.rb

```diff
@@ -7,16 +7,14 @@ class StatisticsControllerTest < ActionController::TestCase
     end
 
     test 'user with viewer rights should succeed in viewing statistics' do
-      @request.session[:user] = users(:one).id
-      users(:one).roles = [Role.default, Role.find_by(name: 'Viewer')]
-      get :index, session: set_session_user
+      setup_user('view', 'statistics')
+      get :index, session: set_session_user(:one)
```

Comment:
This is the main change - `setup_user` adds the nonexistent permissions and then we can do the request as user `:one` as the title of the test suggests, instead of an admin user (default)